packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -169,9 +174,44 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _motionManager = [[CMMotionManager alloc] init];
   [_motionManager startAccelerometerUpdates];
 
-  if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
-    return nil;
+  NSError *outError = nil;
```

Comment:
This is a very confusing name for a method that has an output error already. If you have to keep it, it should be given a name specific to the call, but you could just re-use `localError`.

Or even better, is there a reason these aren't all just directly passing `error` through? It seems like the case above could just be:
```objc
AVCaptureConnection *connection = [self createConnection:error];
if (!connection) {
  return nil;
}
```

and similarly for the new code. The code above looks stylistically wrong to me already, since

>> When dealing with errors passed by reference, it’s important to test the return value of the method to see whether an error occurred, as shown above. Don’t just test to see whether the error pointer was set to point to an error.

https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/ErrorHandling/ErrorHandling.html

Reply:
`NSError *outError = nil;` eliminated in media-settings related PR

Relating your correct suggestion about eliminating `localError`
```
AVCaptureConnection *connection = [self createConnection:error];
if (!connection) {
  return nil;
}
```
is not related to media-settings PR, and there are actually some bug prone constructs, like setting `*error` on unchecked nullable pointer. And at this moment `connection` is actually always `nil` in any test, because not neither mocked nor DI-ed anywhere.